[Mailer] Add event subscriber example to mailer documentation (#16744)
symfony/symfony-docs: mailer.rst

Thanks to this, instead of being delivered immediately, messages will be sent to
the transport to be handled later (see :ref:`messenger-worker`).

Events
------

MessageEvent
~~~~~~~~~~~~

> The MessageEvent allows you to manipulate the send message::

Author: ```suggestion
The MessageEvent allows the transformation of a Message and the Envelope before the email is sent:
```

> The MessageEvent allows the transformation of a Message and the Envelope before the email is sent::

Author: copied from: https://github.com/symfony/symfony/blob/6.1/src/Symfony/Component/Mailer/Event/MessageEvent.php#L19

    use Symfony\Component\EventDispatcher\EventSubscriberInterface;
    use Symfony\Component\Mailer\Event\MessageEvent;
    use Symfony\Component\Mime\Email;

>     class MailerListener implements EventSubscriberInterface

Reviewer: ```suggestion
    class MailerSubscriber implements EventSubscriberInterface
```
It would be more interesting here to change `MailerListener` to `MailerSubscriber` because the example implements the interface `EventSubscriberInterface`

Thanks to this, instead of being delivered immediately, messages will be sent to
the transport to be handled later (see :ref:`messenger-worker`).

Events
------

The mailer component also throws a events:

MessageEvent
~~~~~~~~~~~~

The MessageEvent allows you to manipulate the send message:

> .. code-block:: php

Author: ```suggestion
The MessageEvent allows you to manipulate the send message::
```
